app/src/main/java/com/nextcloud/talk/chat/ChatActivity.kt

```diff
@@ -2514,57 +2503,78 @@ class ChatActivity :
             return
         }
 
-        if (currentConversation!!.remoteServer != null) {
+        if (currentConversation!!.remoteServer?.isNotEmpty() == true) {
             val apiVersion = ApiUtils.getSignalingApiVersion(conversationUser!!, intArrayOf(ApiUtils.API_V3, 2, 1))
             ncApi.getSignalingSettings(
                 credentials,
                 ApiUtils.getUrlForSignalingSettings(apiVersion, conversationUser!!.baseUrl, roomToken)
-            ).blockingSubscribe(object : Observer<SignalingSettingsOverall> {
-                override fun onSubscribe(d: Disposable) {
-                    // unused atm
-                }
-
-                override fun onNext(signalingSettingsOverall: SignalingSettingsOverall) {
-                    if (signalingSettingsOverall.ocs!!.settings!!.externalSignalingServer == null ||
-                        signalingSettingsOverall.ocs!!.settings!!.externalSignalingServer?.isEmpty() == true
-                    ) {
-                        return
+            )
+                .subscribeOn(Schedulers.io())
+                ?.observeOn(AndroidSchedulers.mainThread())
+                ?.subscribe(object : Observer<SignalingSettingsOverall> {
+                    override fun onSubscribe(d: Disposable) {
+                        disposables.add(d)
```

Comment:
Just to be sure, that means that if the signaling settings are received once the activity was closed `onNext` will not be executed (so the Websocket instance will not be got, not the room joined, nor the listeners added), is that correct? If not then that should be somehow prevented, but I guess that is exactly the purpose of the disposable :-)

Reply:
yes that's the purpose. won't be necessary anymore soon when we switch from rxJava to coroutines